src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -129,7 +140,12 @@ internal Binder Binder
             SyntaxToken identifierToken,
             CSharpSyntaxNode context)
         {
-            return new OutLocalSymbol(containingSymbol, scopeBinder, enclosingBinderOpt, typeSyntax, identifierToken, context);
+            if (typeSyntax.IsVar)
+            {
```

Comment:
This test is redundant with a test already performed by an OutLocalSymbol, and this approach is inconsistent with the approach used for other locals. It would be best to have a single design pattern here. Please revert.